renderdoc/core/image_viewer.cpp

```diff
@@ -637,6 +637,7 @@ void ImageViewer::RefreshFile()
   size_t datasize = 0;
 
   bool dds = false;
+  read_dds_data read_data = {};
```

Comment:
I'm going to be annoying and say that I think this is the right approach but it should be refactored so we're not quite so silently hijacking the dds path. This code has been DDS specific because historically DDS files are the only ones with mips, slices, or complex formats, so if we're adding a new format we should untangle that more clearly.

What I'd suggest is:

* Rename `dds` -> `tex` in the structures, add a comment that this is for 'texture' type images - ie. any file format that is more complex in some dimension than one rectangle of pixels in a regular Nbpp float or normalised format.
* `is_dds_file`/`load_dds_from_file`/`write_dds_to_file` can stay named the same.
* The header is a bit annoying, there's not a great 'common' place to put the structs. I think I'd slightly prefer adding a new tiny header in the same place for the structs, but I'd also be fine just renaming the header file similarly `dds` -> `tex` and accept that the dds functions will live there for now.
* Refactor the control flow here such that the 'dds' path isn't just for dds files.

Reply:
Yeah I also had a "ick, this is hijacking structs that were dds only before", but I wondered whether a larger "refactor" would be good or not. Will do!

Reply:
It's definitely borderline. It doesn't need a huge refactor I think mostly I'm picturing a rename pass to be clearer so people reading the code have an easier time and don't make bad assumptions, but otherwise the end result probably won't look that different to what you have here.

Reply:
Ok, I think I did your suggestion, and factored out all the EXR loading into a separate function, since it was getting fairly large being just inside RefreshFile. That ok?

Reply:
Certainly a reasonable step, and looks good thanks!